filter_test.go

```diff
@@ -47,13 +47,17 @@ func TestFilterStructs(t *testing.T) {
 	require.Len(t, filtered, 2)
 }
 
+type filterContextKey string
```

Comment:
I think redefining the contextKey for each specific type doesn't add anything, just more clutter. I suggest just using `contextKey` for the type (already defined in common.go)